File: sportsbook-app/payment-validator-service/src/main/java/com/sportsbook/payment/validator/ports/CreditCheckPort.java
package com.sportsbook.payment.validator.ports;

import com.sportsbook.payment.contracts.DepositEnrichedEvent;
import com.sportsbook.payment.contracts.FundingSourceType;

public interface CreditCheckPort {
    FundingSourceType checkSource(DepositEnrichedEvent depositEnriched);
}

File: sportsbook-app/payment-validator-service/src/main/kotlin/com/sportsbook/payment/validator/ports/CreditCheckPort.kt
package com.sportsbook.payment.validator.ports

import com.sportsbook.payment.contracts.DepositEnrichedEvent
import com.sportsbook.payment.contracts.FundingSourceType

interface CreditCheckPort {
    fun checkSource(depositEnriched: DepositEnrichedEvent): FundingSourceType
}
